File: src/main/java/org/polyfrost/overflowanimations/hooks/TransformTypeHook.java
package org.polyfrost.overflowanimations.hooks;

import net.minecraft.client.renderer.block.model.ItemCameraTransforms;

import java.util.EnumSet;

public final class TransformTypeHook {
    public static ItemCameraTransforms.TransformType transform;
    private static final EnumSet<ItemCameraTransforms.TransformType> ITEM_CAMERA_TYPES = EnumSet.of(ItemCameraTransforms.TransformType.GROUND, ItemCameraTransforms.TransformType.FIXED);

    private TransformTypeHook() {
    }

    public static boolean shouldBeSprite() {
        return shouldNotHaveGlint() || isRenderingInGUI();
    }

    public static boolean isRenderingInGUI() {
        return transform == ItemCameraTransforms.TransformType.GUI;
    }

    public static boolean shouldNotHaveGlint() {
        return ITEM_CAMERA_TYPES.contains(TransformTypeHook.transform);
    }
}

File: src/main/kotlin/org/polyfrost/overflowanimations/hooks/TransformTypeHook.kt
package org.polyfrost.overflowanimations.hooks

import net.minecraft.client.renderer.block.model.ItemCameraTransforms.TransformType
import java.util.*

object TransformTypeHook {

    var transform: TransformType? = null

    private val cameraTypes: EnumSet<TransformType?> = EnumSet.of(
        TransformType.GROUND,
        TransformType.FIXED
    )

    fun shouldBeSprite(): Boolean {
        return shouldNotHaveGlint() || isRenderingInGUI
    }

    private val isRenderingInGUI: Boolean
        get() = transform == TransformType.GUI

    fun shouldNotHaveGlint(): Boolean {
        return cameraTypes.contains(transform)
    }
}
